### PR TITLE
Make player theme selection more intuitive & straightforward

### LINKED ISSUE #123: Make player theme selection more intuitive & straightforward
### Operating system
Linux Mint 20.2 Cinnamon

### Browser name and version
Firefox 94.0

### Expected behavior
"Auto" theme should be one of three equally selectable options, along with "Light" and "Dark", and should be the default.

### Actual behavior
"Auto" theme is a toggle switch, and greys out (but does not deselect) "Light" and "Dark" options when turned on. Switching back to "Light" or "Dark" is not possible until the "Auto" switch is turned off. "Light" theme is the default (and only option when logged out).

### Steps to reproduce behavior
While logged in to the player, open the menu at the upper right hand side. Selecting the "auto" theme, prevents "light" or "dark" from being selected until "auto" is unchecked. These options are greyed out, but circle next to the previously selected option is still filled in, leading to the possibility of an auto dark theme, while a greyed out "light" is still shown as selected, or vice versa.

### PR DESCRIPTION
**Pull request by peterklingelhofer**
This makes `Auto` theme one of three equally selectable options, along with `Light` and `Dark`, and the default, and changes `Auto` from a `checkbox` to a `radio` style input, such as the `Light` and `Dark` already have.

The `Auto` option also now properly detects user system preferences, and if the user previously had `Auto` selected (detected via `localStorage`).

If the user is using a dark mode on their system, it now honors their preferences on the initial page load. Previously, for users whose computers are in dark mode (at least on my system), on the initial page load, the website would appear light, even though it appears that `Auto` is the default. When I would toggle `Auto` off and on again (and for all subsequent toggles) when it was a checkbox, the website would appear dark, and seemed to have figured out the user's system preference by then. These changes fix that, and now the behavior for `Auto` is consistent between the initial default `Auto` and subsequent theme changes (switching off of `Auto` and then back, etc). 

I wanted to make sure that the initial load of `Auto` defaulting to light mode was a indeed a bug that I fixed and not intentional, so I've included ample screenshots below to keep everyone on the same page as to how exactly this functionality is being adjusted.

This closes https://github.com/resonatecoop/stream/issues/123.

**Screenshot walk-through of all of the different possible scenarios:**
Light system user is logged out, website defaults to `Auto`
light-system-user-logged-out

Light system user logs in, website defaults to `Auto`
light-system-user-logged-in-auto

Light system user logs in, then switches to `Light` theme
light-system-user-logged-in-light

Light system user logs in, then switches to `Dark` theme
light-system-user-logged-in-dark

Dark system user is logged out, website defaults to `Auto`
dark-system-user-logged-out

Dark system user logs in, website defaults to `Auto`
dark-system-user-logged-in-auto

Dark system user logs in, then switches to `Light` theme
dark-system-user-logged-in-light

Light system user logs in, then switches to `Dark` theme
dark-system-user-logged-in-dark

### CODE REVIEW

File: beta/components/theme-switcher/index.js [hunk 1]
```diff
@@ -84,6 +84,16 @@ class ThemeSwitcher extends Component {
   load (el) {
     this.local.auto = localStorage !== null && !!localStorage.getItem('color-scheme-auto')
     this.rerender()
+    const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
```
Comment (auggod) [inline beta/components/theme-switcher/index.js]: Semicolon should be removed

Comment (auggod) [reply beta/components/theme-switcher/index.js]: I would also check if matchMedia is available

Comment (peterklingelhofer) [reply beta/components/theme-switcher/index.js:87]:
```suggestion
    if (
      window !== undefined &&
      window.matchMedia('(prefers-color-scheme: dark)') !== undefined &&
      window.matchMedia('(prefers-color-scheme: dark)').matches !== undefined
      ) {
        const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches
```

Would adding the above checks be suitable for checking if `matchMedia` is available? The above suggestion would of course also need the closing bracket for the new if block below `this.local.machine.emit('theme:toggle')`.

Logging these three checks in the console appear to consistently yield `true` in my testing:
```js
console.log(window !== undefined)
console.log(window.matchMedia('(prefers-color-scheme: dark)') !== undefined)
console.log(window.matchMedia('(prefers-color-scheme: dark)').matches !== undefined)
```

Comment (peterklingelhofer) [reply beta/components/theme-switcher/index.js]: I've addressed your suggestions here: https://github.com/resonatecoop/stream/pull/128/commits/3bf0f13687122e596000945bcb921239bfdb8839.

Let me know if some of these `!== undefined` checks I've added in the `if` block are excessive, if so, I'm happy to remove them.

**Comment by peterklingelhofer** [comment]
It looks like the linter for this project prefers operators at the end of the line, so I've removed the last commit that moved those to the beginning.

**Comment by auggod** [comment]
I recommend to install standard and not commit anything that does not pass its validation checks.
Usage
```npm install standard -g```

```standard --fix myfile.js```

**Comment by peterklingelhofer** [comment]
> I recommend to install standard and not commit anything that does not pass its validation checks.
> 
> Usage
> 
> ```npm install standard -g```
> 
> 
> 
> ```standard --fix myfile.js```
> 
> 

Perfect, I will do this for all future commits. Thank you!

**Comment by auggod** [comment]
Thank your for taking the time to improve our player!

**Pull request merged**